xbmc/video/VideoDatabase.cpp

```diff
@@ -6518,6 +6518,10 @@ bool CVideoDatabase::GetEpisodesByWhere(const std::string& strBaseDir, const Fil
         pItem->SetOverlayImage(CGUIListItem::ICON_OVERLAY_UNWATCHED, movie.m_playCount > 0);
         pItem->m_dateTime = movie.m_firstAired;
         pItem->GetVideoInfoTag()->m_iYear = pItem->m_dateTime.GetYear();
+        if (movie.m_iSpecialFlag == EPISODE_FLAG_SEASON_FINALE)
+          pItem->SetProperty("season_finale", "1");
+        else if (movie.m_iSpecialFlag == EPISODE_FLAG_SERIES_FINALE)
```

Comment:
Since only one flag is stored (which is enough imo) you should check for series finale first. Otherwise the series flag might be eaten in case some future API returns both.